### ISSUE TITLE
AGPL / LGPL

### ISSUE THREAD

**Issue by smessmer** [mdbootstrap/TW-Elements#1936]
I have started to use Tailwind Elements in a new website I'm building, and in the process of doing so, I have started to build [leptos-twelements](https://github.com/smessmer/leptos_twelements) which is an open source integration of Tailwind Elements into the [Leptos](https://leptos.dev/) web framework. I really like the UI design of Tailwind Elements. It looks beautiful. Thank you for building this great set of components.

But I noticed that Tailwind Elements uses the AGPL license and that is a quite restrictive one. [Google interprets the AGPL](https://opensource.google/documentation/reference/using/agpl-policy) to mean that any code linking to it must be open source as well, i.e. any website using Tailwinds Elements would have to open source all of their website code, possibly even the backend code.

This would make Tailwind Elements unusable for most websites and because I don't want to open-source my website code, it would likely mean I'll have to move to one of the other component libraries, e.g.
- https://www.tailwindawesome.com/?type=kit
- https://blog.logrocket.com/10-best-tailwind-css-component-libraries/
- https://stackdiary.com/tailwind-components-ui-kits/

I don't believe I would be the only one.

Have you considered changing the license to something like the LGPL? That would still mean that any changes to Tailwind Elements itself are copylefted (e.g. if somebody creates variants of Tailwind Elements components, they'll have to be open source), but it wouldn't force websites themselves to make all of their code open source. Or maybe Apache/MIT like Tailwind CSS itself? I think a more permissive license would greatly increase the number of users who would consider Tailwind Elements.

**Comment by filipkappa** [comment]
Hello,

Thank you for expressing your concerns.

We believe that the AGPL is more in line with the open-source spirit, which is focused on **sharing knowledge** and **driving innovation**. 

Of course, we understand that the AGPL may not be the best fit for some specific use cases, which is why we have implemented a **dual licensing** model.

This allows us to offer Tailwind Elements under the AGPL for those who share our commitment to open-source values, while also offering a **custom license** for those who need a **more permissive license for their specific use case**.

Tailwind Elements is meant to be a community-driven project, and we hope this decision will help to enrich the wider open-source community as well, by requiring most of non-sensitive innovations to be shared.

---

In summary, AGPL is our default license for the package downloaded directly from our site.
If you need a different license, we are open to issue it to you individually - based on your specific needs :)

Check out the **custom license** form here:
https://tailwind-elements.com/license/

Keep coding!
